### ISSUE TITLE
 Explanation of the input parameters of the forward function of the model

### ISSUE THREAD

**Issue by code10086web** [gabeur/mmt#19]
Thanks a lot for sharing such great work. 
In the process of reading the code, there are some questions that cannot be understood. I would like to ask if you can give me some tips. Thank you very much.

Q:  Explanation of the input parameters of the forward function on line 312 of file model/model.py. It’s really hard for me to understand the meaning of parameter **features_t** and parameter **features_ind**. Looking forward to your suggestions.

![image](https://user-images.githubusercontent.com/53412966/132129551-bc86ef4c-b81e-42ff-bd27-7a68b9cfac73.png)

**Comment by gabeur** [comment]
Thanks for your interest in our work.
- features_t encode the time in the video at which each feature was extracted. As described in the section 3.1 of the paper, we use temporal embeddings to provide temporal information about the time in the video when each feature was extracted. The temporal information of each feature is discretized [here](https://github.com/gabeur/mmt/blob/1c2ffaa68da1df813d96fff51af1c29a5b0544d2/model/model.py#L519) and then embedded [here](https://github.com/gabeur/mmt/blob/1c2ffaa68da1df813d96fff51af1c29a5b0544d2/model/bert.py#L98).
- features_ind encode if the feature is valid (1) or if it is padding (0). It is then used [here](https://github.com/gabeur/mmt/blob/1c2ffaa68da1df813d96fff51af1c29a5b0544d2/model/model.py#L557) to set the attention mask so that padding tokens are not attended.

**Comment by code10086web** [comment]
Thank you very much for your reply. By any chance, Is it possible for you to explain the meaning of the parameters of the **vid_bert**  function on line 577 of file model/model.py?  especially the parameter **input_ids**.

![image](https://user-images.githubusercontent.com/53412966/132222244-cf63626c-6605-4657-91f4-a6ef59065bab.png)

**Comment by gabeur** [comment]
- input_ids were used initially to encode the different token classes but it became useless with the temporal embeddings. You will find in the [forward function](https://github.com/gabeur/mmt/blob/1c2ffaa68da1df813d96fff51af1c29a5b0544d2/model/bert.py#L87) that it is only used to infer the shape of the missing token_type_ids.
- attention_mask encode the tokens that should be attended (valid tokens) in contrast to the padding tokens that should not be attended.
- token_type_ids encode the expert used to extract the feature (called "Expert embeddings" in part 3.1 of [the paper](https://arxiv.org/pdf/2007.10639.pdf)).
- position_ids encode the time in the video at which each feature was extracted (called "Temporal embeddings" in part 3.1 of [the paper](https://arxiv.org/pdf/2007.10639.pdf))
- features are the expert features (called "Features" in part 3.1 of [the paper](https://arxiv.org/pdf/2007.10639.pdf))
